### ISSUE TITLE
Script could not run on OSX

### ISSUE THREAD

**Issue by blkwinds** [renatosilva/easyoptions#12]
OSX use its own sed etc.
Without install gnu sed, the script could not run. It is a good product. I really hope I can have it on OSX without install additional packages.

**Comment by fatso83** [comment]
See #9 - it has fixes for this. The fix is really trivial, so you can simply fork the project and fix it yourself. Or simpler, just use the fork from the PR :blush:

**Comment by kurahaupo** [comment]
Or better, make a new pull request that uses POSIX sed instead of
extensions.

**Comment by fatso83** [comment]
@kurahaupo Maybe it's time to just simply join forces around a new fork where we can integrate the various suggestions and pull requests? I mean, no issue has been closed for 4 years and no pull request has ever been merged ...  It would necessarily need to go hand in hand with a new Gem as well, but I haven't really dabbled with Ruby enough to do that efficiently.

**Comment by kurahaupo** [comment]
@fatso83 Yes that sounds like a plan. (I should have replied much earlier, sorry about that.)

Personally I've not been using this module as it is missing some key features that I need, and adding them will involve a rather tortuous upgrade path.

I think that the next step is to change the mechanism for passing the template to the shell module, so that instead of having to find and open the script, it's passed either as a heredoc:

```sh
. easyoptions.shlib --heredoc <<EndOfOptions
 -v             verbose
 -cCount        limit to Count iterations (default $Count)
EndOfOptions
```

or as a multi-line string argument:

```sh
. easyoptions.shlib --arg "
 -v             verbose
 -cCount        limit to Count iterations (default $Count)
"
```

I'm making a specific proposal for other languages, but something closely approximating one of these two methods should be available in most languages.

**Comment by kurahaupo** [comment]
Or perhaps we should just start again, because the _very first_ design decision was a mis-step: parsing structured data out of a readable document is always going to be harder (and less reliable) than generating a readable doc from structured data.

This project began as a learning exercise; good for what it is, but early decisions mean that it's going to be forever awkward. (Mind you, that pretty much describes the shell itself.)

**Comment by PiDockMedia** [comment]
Did any of the ideas here come to fruition? I could just fork and fix it because it is fairly simple, but I am curious if it has been fixed/improved/better recommendations for easy command line switches.

**Comment by fatso83** [comment]
Nope.

**Comment by kurahaupo** [comment]
I suggest going ahead and forking.

It seems that @renatosilva vanished from Github - no commits, issues, or comments since July 2016.
